doc/lua_api.txt

```diff
@@ -641,6 +674,71 @@ Result is more like what you'd expect if you put a color on top of another
 color, meaning white surfaces get a lot of your new color while black parts
 don't change very much.
 
+A Multiply blend can be applied between two textures by using the overlay
+modifier with a brightness adjustment:
+
+    textureA.png^[contrast:0:-64^[overlay:textureB.png
+
+#### `[screen:<color>`
+
+Apply a Screen blend with the given color. A Screen blend is the inverse of
+a Multiply blend, lightening images instead of darkening them.
+
+`<color>` is specified as a `ColorString`.
+
+A Screen blend can be applied between two textures by using the overlay
+modifier with a brightness adjustment:
+
+    textureA.png^[contrast:0:64^[overlay:textureB.png
+
+#### `[hsl:<hue>:<saturation>:<lightness>`
+
+Adjust the hue, saturation, and lightness of the texture. Like
+"Hue-Saturation" in GIMP, but with 0 as the mid-point.
+
+`<hue>` should be from -180 to +180
+
+`<saturation>` and `<lightness>` are optional, and both from -100 to +100
+
+#### `[contrast:<contrast>:<brightness>`
+
+Adjust the brightness and contrast of the texture. Conceptually like
+GIMP's "Brightness-Contrast" feature but allows brightness to be wound
+all the way up to white or down to black.
+
+`<contrast>` is a value from -127 to +127.
+
+`<brightness>` is an optional value, from -127 to +127.
+
+If only a boost in contrast is required, an alternative technique is to
+hardlight blend the texture with itself, this increases contrast in the same
+way as an S-shaped color-curve, which avoids dark colors clipping to black
+and light colors clipping to white:
+
+    texture.png^[hardlight:texture.png
+
+#### `[overlay:<file>`
+
+Applies an Overlay blend with the two textures, like the Overlay layer mode
+in GIMP. Overlay is the same as Hard light but with the role of the two
+textures swapped, see the `[hardlight` modifier description for more detail
+about these blend modes.
```

Comment:
So `a.png^[overlay:b.png` is the same as `b.png^a.png`?
Not sure I get this

Edit: Reading on, yeah nope it isn't. Could this be explained better for people who are not familiar with anything?

Reply:
>> like the Overlay layer mode in GIMP
>
> So a.png^[overlay:b.png is the same as b.png^a.png?

later you suggest leaving out comments like "like XXX in GIMP", but the above confusion along with appgurueu's [suggestion that brightness should be seperated](https://github.com/minetest/minetest/pull/10100#discussion_r704519671) makes me think I'm not hammering on this drum hard enough, and it's not merely for lua modders to understand.

"like XXX in GIMP" could well be the wrong way to say it. What I'm trying to convey is there's a time-tested well thought out standard set of colour operations which are universal across CSS and every graphics program (GIMP, Photoshop, Affinity, Paint.Net etc. - all of them), they do it that way for smart reasons, and consequently they're also now the colour operations artists are familiar with. This PR is attempting to expose some of those operations in the modding API.

My implementations may leave room for improvement, e.g. someone might think [linear light](https://github.com/minetest/minetest/pull/10100#issuecomment-650471088) is worthwhile, or wonder why I used a tan call in apply_brightness_contrast(), or suspect that there's a bug in irrlicht's HSL code related to increasing saturation of greys (I suspect this, but haven't yet investigated). In each case it should be clear that unlike other modifiers, this code doesn't define what these texture modifiers are supposed to do, their definitions and answers to what/why/wtf are outside of minetest and in the wider industry. 

You're right that it's not clear, but I might not be the right person to make it clearer. I'll have to think more.

Reply:
perhaps I could change the first line to

`Applies [an Overlay blend](https://en.wikipedia.org/wiki/Blend_modes#Overlay) with the two textures, like the Overlay layer mode`

Hard to read as a text file but it might be a markdown file one day